app/main.py

```diff
@@ -1,6 +1,22 @@
+from functools import wraps
 from typing import Callable
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> None:
+        func_key = func.__name__
+        key = (func_key, args, frozenset(kwargs.items()))
+
+        if key in cache_dict:
+            print("Getting from cache")
+            return cache_dict[key]
+        else:
```

Comment:
There is still no need for `else`, the code will not execute after the `return` statement. Remove it from the code, please.